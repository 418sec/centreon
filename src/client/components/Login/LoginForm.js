import React, { Component } from 'react'
import Typography from 'material-ui/Typography'
import Paper from 'material-ui/Paper'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'
import TextField from 'material-ui/TextField'
import Button from 'material-ui/Button'

const styles = {
  root: {
    position: 'relative'
  },
  loginButton: {
    background: 'linear-gradient(-45deg, #009fdf 30%, #0072ce 90%)',
    width: '100%',
    borderRadius: 3,
    border: 1,
    color: 'white',
    height: 42,
    marginTop: '30px',
    padding: '0 30px',
  },
  errorContainer: {
    position: 'absolute',
    top: '0'
  },
  errorText: {
    color: '#E00B2D',
    margin: '0px 14px'
  },
  loginTitle: {
    marginTop: '4rem'
  },
  card: {
    maxWidth: 345,
    boxShadow: 'none',
    backgroundColor: 'transparent'
  }
};

/**
 * Login Form Component
 */
class LoginForm extends Component {
  constructor (props) {
    super(props)
    this.state = {
      user: {
        username: this.props.user.username,
        password: this.props.user.password
      }
    }
  }

  onChange = (event) => {
    const field = event.target.name
    const user = this.state.user
    user[field] = event.target.value

    this.setState({
      user
    })
  }

  onSubmit = (event) => {
    event.preventDefault()
    this.props.onSubmit(this.state.user)
  }

  render () {
    const { classes } = this.props
    return (
      <div className={classes.root}>
        <Paper className={classes.errorContainer} >
        </Paper>
        <Typography type="headline" align="center">
          SIGN IN
        </Typography>
          <form onSubmit={this.onSubmit} >
            <TextField
              fullWidth
              placeholder="Your Centreon username"
              id="username"
              name="username"
              label="username"
              value={this.state.user.username}
              onChange={this.onChange}
              margin="normal"
            />
            <TextField
              fullWidth
              placeholder="Your Centreon password"
              id="password"
              label="Password"
              name="password"
              type="password"
              value={this.state.user.password}
              onChange={this.onChange}
              autoComplete="current-password"
              margin="normal"
            />
            <Button raised color="primary" type="submit" className={classes.loginButton}>Log in </Button>
          </form>
      </div>
    )
  }
}

/**
 * Login Form Prototypes
 */
LoginForm.propTypes = {
  /**
   * LoginForm css classes
   * @param {Object} classes
   */
  classes: PropTypes.object.isRequired,
  /**
   * Centreon LDAP User credentials
   *
   */
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired
  }),
  /**
   * Authenticate with credentials
   */
  onSubmit: PropTypes.func.isRequired
}

LoginForm.defaultProps = {
  user: {
    username: '',
    password: ''
  }
}

export default withStyles(styles)(LoginForm)
