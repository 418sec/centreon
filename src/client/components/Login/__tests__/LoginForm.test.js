/* eslint-env jest */
import React from 'react';
import Enzyme from 'enzyme';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import LoginForm from '../LoginForm';

Enzyme.configure({ adapter: new Adapter() });

describe('LoginForm', () => {
  it('renders auth form correctly', () => {
    const login = shallow(
      <LoginForm
        user={{ username: 'user1', password: 'password1' }}
        onSubmit={() => {}}
      />,
    );
    expect(toJson(login)).toMatchSnapshot();
  });

  it('submit form', () => {
    let userSubmitted;

    const login = mount(
      <LoginForm
        user={{ username: 'user1', password: 'password1' }}
        onSubmit={(user) => { userSubmitted = user; }}
      />,
    );

    login.find('form').simulate('submit');
    expect(userSubmitted).toEqual({
      username: 'user1',
      password: 'password1',
    });
  });

  it('change form', () => {
    let userChanged;
    const login = mount(
        <LoginForm
          user={{ username: 'user0', password: 'password0' }}
          onSubmit={(user) => { userChanged = user; }}
        />,
    );

    login.find('input#username').simulate('change', {target: {name: 'username', value: 'user1'}})
    login.find('input#password').simulate('change', {target: {name: 'password', value: 'password1'}})
    login.find('form').simulate('submit')

    expect(userChanged).toEqual({
      username: 'user1',
      password: 'password1',
    });
  });
});