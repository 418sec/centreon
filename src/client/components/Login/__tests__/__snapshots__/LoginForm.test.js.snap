// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders auth form correctly 1`] = `
<LoginForm
  classes={
    Object {
      "card": "LoginForm-card-6",
      "errorContainer": "LoginForm-errorContainer-3",
      "errorText": "LoginForm-errorText-4",
      "loginButton": "LoginForm-loginButton-2",
      "loginTitle": "LoginForm-loginTitle-5",
      "root": "LoginForm-root-1",
    }
  }
  onSubmit={[Function]}
  user={
    Object {
      "password": "password1",
      "username": "user1",
    }
  }
/>
`;
